lib/engine/auto_router.rb

```diff
@@ -34,21 +34,26 @@ def compute(corporation, **opts)
       # Add a per-node timeout to ensure we don't spend ALL time on first node's paths in huge maps,
       # which can cause all train routes to conflict with each other
       node_timeout = path_timeout / nodes.size
+      path_walk_timed_out = false
 
       now = Time.now
 
       skip_paths = static.flat_map(&:paths).to_h { |path| [path, true] }
+      # if only routing for subset of trains, keep track of which ones we'll assemble routes for
+      skip_trains = static.flat_map(:train).to_a
+      look_trains = trains - skip_trains
```

Comment:
can you jsut do trains = trains - skip_trains

Reply:
Sure

Reply:
Actually it turns out I can't, there's some weird crossover side effect afterwards when I click a train to clear it (an exception where '5' is not a train).  It's like `trains = trains - skip_trains` is modifying the actual train collection in the RR company.